zilliqa/src/db.rs

```diff
@@ -584,6 +726,63 @@ impl Db {
     }
 }
 
+pub fn snapshot_block_with_state<P: AsRef<Path> + Debug>(
```

Comment:
Could we document the snapshot file format separately? As it stands, the only way to work it out is by reading both the serialisation and deserialisation code.

Reply:
I've considered this but I feel like the only people who would want to manually edit them would also be able to read the code.
However since I've considered it and you think it's also useful then maybe worth doing after all. Any preference on where to put it? I'm thinking just sticking a new file under `docs/`.

Reply:
It's going to be very hard to review PRs that touch either ser or de code for checkpoints without knowing what the intended format is. docs/ works for me. See ^^^ for a suggestion that might help with this as well (you can then use the rust struct defn you give to serde as a spec).